### PR TITLE
bug: fix spin nlist in spin_model

### PR DESCRIPTION
**Pull request by iProzd**
## Summary by CodeRabbit

- **New Features**
	- Expanded backend support to include PyTorch and DP in the energy spin training documentation.
	- Updated spin settings for TensorFlow and PyTorch/DP in the energy spin training documentation.
	- Added new sections for loss functions and data preparation in the energy spin training documentation.

- **Bug Fixes**
	- Adjusted test conditions and initialization parameters in various model tests to align with updated functionalities.

- **Tests**
	- Increased selection values in permutation tests to enhance test coverage and reliability.

### CODE REVIEW

**Comment by iProzd** [comment]
@hztttt Please review this PR, which fixes the nlist bugs in DPA2. (It seems I can not explicitly assign you as a reviewer.)

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

**Review by hztttt — APPROVED** [review]

**Comment by iProzd** [comment]
Two bugs:
1. need buffer region for spin nlist.
2. sel is too small for **dpa2 with spin** (rcut 6.0 with sel 30), where they may sometimes select different self images on the bondary with the same distances (which may cause the same `g` but different `r` during `grrg` update). When there's no spin, sel needed is half less that, so it was ok.

**Comment by iProzd** [comment]
https://github.com/deepmodeling/deepmd-kit/pull/3718/commits/c91db5256c68af79e936d4030c1c7419343532dc adds docs for spin in PyTroch/DP. But the data format (old/new) stills need a detail doc, i've opened a issue (#3760 ) about that. Need help from @hztttt . Maybe in a seperate PR.

**Review by wanghan-iapcm — APPROVED** [review]
